specification/compute/resource-manager/Microsoft.Compute/preview/2020-09-30/sharedGallery.json

```diff
@@ -0,0 +1,1040 @@
+              "tenant"
+            ],
+            "x-ms-enum": {
+              "name": "SharedToValues",
+              "modelAsString": true
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SharedGalleryList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a gallery.": {
+            "$ref": "./examples/ListSharedGalleries.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/sharedGalleries/{galleryUniqueName}/images": {
+      "get": {
+        "tags": [
+          "SharedGalleryImages"
+        ],
+        "operationId": "SharedGalleryImages_List",
+        "description": "List shared gallery images by subscription id or tenant id.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationNameParamter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "galleryUniqueName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The unique name of the Shared Gallery."
+          },
+          {
+            "name": "sharedTo",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The query parameter to decide what shared galleries to fetch when doing listing operations.",
+            "enum": [
+              "tenant"
+            ],
+            "x-ms-enum": {
+              "name": "SharedToValues",
+              "modelAsString": true
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SharedGalleryImageList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get a gallery.": {
+            "$ref": "./examples/ListSharedGalleryImages.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/locations/{location}/sharedGalleries/{galleryUniqueName}/images": {
```

Comment:
galleries are shared with a subscription or tenant.  When would a resource group be in the picture?

Reply:
This is for returned id. The returned resource will be granted with 'id' for deployment or other use. 'Resource group name' can help with that aspect.

Reply:
How would a sharedGallery get placed in a resource group?  There is nowhere during sharing where the sharer gets to specify the RG and it shouldn't be placed in a random RG.  There is no requirement that it must be in a resource group to be referenced from other resources.

Reply:
The concept of subscriptionId, tenantId and resourceGroup is different from previous apis. For the shared galleries, the sub id and resource group input means the caller's sub id and resource group. And they do not own the shared gallery so shared galleries do not need to be put into resource group. We use want to put the caller's info in the returned id.

Reply:
That doesn't make sense.  Either it exists in a resource group or it doesn't.  There is no requirement that it needs to be faked to appear in a resource group.  Just expose it at subscription level (where it was shared to) and let users reference it from there

Reply:
If a gallery is shared to a sub, it will be shared to all resource groups under this sub automatically(in your word, exist in the resource group). And let's say for the caller, some caller is under some resource group and the returned id for them should show 'resource group', which indicates that they can access the shared gallery from this resource group(some users' activity may be limited inside some resource group). And some caller is not under any resource group but under the shared subscription, and we return id which is only on the subscription level for them and do not contain 'resource group'.

Reply:
The resource ID should be the resource ID, it shouldn't change depending on where the user makes the call from.  Listing in a resource group can, and should, show everything shared with the subscription the resource group is in and the resources returned from that API should use the subscription-level resource ID.